src/API/Site/Controllers/Jetpack/AccountController.php

```diff
@@ -189,11 +189,17 @@ protected function get_connected_callback(): callable {
 
 	/**
 	 * Determine whether Jetpack is connected.
+	 * Check if manager is active and we have a valid token.
 	 *
 	 * @return bool
 	 */
 	protected function is_jetpack_connected(): bool {
-		return $this->manager->is_active();
+		if ( ! $this->manager->is_active() ) {
+			return false;
+		}
+
+		$token = $this->manager->get_tokens()->get_access_token( false, false, false );
+		return $token && ! is_wp_error( $token );
```

Comment:
❓ / 💅  

Why not just `return $this->manager->get_tokens()->get_access_token();`

Reply:
Nice spot, I didn't look at the function close enough to realize that with the default parameters it will suppress the errors.